camel/functions/openai_function.py

```diff
@@ -245,7 +179,6 @@ def get_openai_tool_schema(self) -> Dict[str, Any]:
         Returns:
             Dict[str, Any]: The OpenAI tool schema for this function.
         """
-        self.validate_openai_tool_schema(self.openai_tool_schema)
```

Comment:
Why did we validate `openai_tool_schema` during `get` but not `set` in the previous version?

Reply:
> Why did we validate `openai_tool_schema` during `get` but not `set` in the previous version?

@dandansamax  Sry I have no idea. I only add `pydantic` v1 support to this function before. The main logic was done in #393 . See https://github.com/camel-ai/camel/pull/393/files#diff-5ef2fd2f375255947f98e1a9359dd96d48d9e548b32312be896ab05dd47e9b17R242-R253.

Reply:
Thanks @ocss884. Let's call the author. Hi @forever-ly, do you have some ideas?